test/test_data_collector.py

```diff
@@ -69,6 +72,10 @@ def test_timer_error_handling():
 
 
 def test_data_writing(tmp_path):
+    def validate_json(filename: Path) -> bool:
```

Comment:
IDKY but this type conversion makes me uncomfortable.
Take a look at the conversion table [here](https://docs.python.org/3/library/json.html#json-to-py-table).

I'd prefer that this function is either:
1. renamed so that it explains what it's actually doing (loading a JSON from a file)
2. checks if `load()` actually loads some value/object and/or doesn't throw.

Reply:
How about something like `def load_json_to_validate(...)`?

Reply:
Probably... At least also have a check to make sure it doesn't throw.

Reply:
I think the implementation looks like this:

```
def validate_json(filename: Path) -> bool:
  try:
    with filename.open() as f:
      json.load(f)
    return True
  except JSONDecodeError:
    return False
```

Reply:
Ah, I see what you mean now. Thanks.